docs/madr/3-handlers.md

```diff
@@ -0,0 +1,112 @@
+# Process Handlers for ANT Registry
+
+- Status: proposed
+- Deciders: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus], [Dylan]
+
+## Context and Problem Statement
+
+To enable the functionality of the ANT registry as outlined in the [Initial ADR], new process handlers need to be defined and implemented. The main goal is to ensure that the registry is always up to date and can efficiently handle ownership queries and integrations with our current (and future) systems.
+
+## Decision Drivers
+
+The main drivers for this decision revolve around:
+
+- **Efficiency**
+  - Optimized lookups for process IDs and wallet addresses.
+- **Scalability**
+  - Ability to handle increasing numbers of ANTs and associated processes.
+- **Maintainability**
+  - Clear, modular handlers for different operations.
+- **Security**
+  - Consideration for preventing abuse and spam in the future.
+
+## Considered Options
+
+The `State-Notice` and `Access-Control-List` handlers are required for storing the ANT owner, however the `Register` handler was discussed deeply to see if it was necessary at all. The names were decided based on the ANT contracts
+
+### With Register
+
+- **Pros:**
+  - Allows triggering other workflows by notifying the ANT with a `Register-Notice`
+  - Can Register ANT's from other workflow without the ANT registering itself.
+- **Cons:**
+  - Increased complexity in management.
+  - Higher initial development effort.
+
+### Without Register
+
+- **Pros:**
+  - Less complex
+- **Cons:**
+  - Increased complexity in management.
+  - Unable to index ANT's without the ANT itself sending the state to the registry, which could be added as a handler to ants, but uneccesary.
+
+## Decision Outcome
+
+It was decided to proceed with the `Register` handler with the caveat that it is not necessarily required for ANT's to add their state to the registry and that the `State-Notice` handler can still accept state to include the ANT.
```

Comment:
Since this was the big question, I think it should be in the title of this ADR, e.g.
` # Provide a "Register" handler in the ANT Registry to provide for permissionless registration of previously undiscovered ANTs`